chart/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if .Values.metrics.enabled -}}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "ots.serviceAccountName" . }}
```

Comment:
Shouldn't this also use `ots.fullname`?

Reply:
If you don't provide a custom name it's falling back to `ots.fullname`: https://github.com/ContinuousSecurityTooling/ots-helm-chart/blob/10-add-support-for-kube-prometheus-stack-monitoring/chart/templates/_helpers.tpl#L75

Reply:
Still why is the `ServiceMonitor` named from the `ots.serviceAccountName` template? It should use the `ots.fullname` template directly.

Reply:
yes it could be, but sometimes people want to bring their own service accounts and use them, so there needs to be the possibility to overwrite the rule...
Therefore I added the helper

Reply:
It's not about the helper. Why is the **ServiceMonitor** named as the **service account**? A ServiceMonitor has **nothing** to do with a service-account. A service account is used in a deployment. The ServiceMonitor is a custom-resource from the KubeProm-Stack, it does not need or use a service account and there is no reason the ServiceMonitor should be named `karl` when I want my service-account to be named `karl`.

```yaml
kind: ServiceMonitor
metadata:
  name: {{ include "ots.serviceAccountName" . }}
```

Reply:
oh god, you‘re right. I will fix that and test it next week in my cluster and make a release

Thanks for pointing to that